query/src/main/java/io/keen/client/java/SavedQueryPut.java

```diff
@@ -0,0 +1,111 @@
+package io.keen.client.java;
+
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+import io.keen.client.java.http.HttpMethods;
+
+/**
+ * Represents PUT requests performed against the Saved/Cached Queries API.
+ *
+ * @author masojus
+ */
+class SavedQueryPut extends SavedQueryRequest {
+    private final KeenQueryRequest query;
+    private final int refreshRate;
+
+    private final Map<String, Object> miscProperties;
+
+
+    SavedQueryPut(String queryName,
+                  String displayName,
+                  KeenQueryRequest query,
+                  int refreshRate,
+                  Map<String, ?> miscProperties) {
+        super(HttpMethods.PUT, true /* needsMasterKey */, queryName, displayName);
```

Comment:
I hate literals as actual parameters when reading code if not in the IDE. In the IDE I can just hover and see what formal parameter that maps to, but reading it elsewhere, like here in Git, it's so much more useful to have the comment for `null` and boolean literals and number type literals, etc. It's something I always do in C and C++ programming, and if there's a good reason not to do it in Java, let me know, but my opinion is that it adds clarity.